adapters/MoPub/MoPubAdapter/GADMediationAdapterMoPub.m

```diff
@@ -0,0 +1,227 @@
+ and 2) non-personally identifiable categories before they are forwarded to MoPub due to GDPR.
+ */
+- (NSString *)getKeywords:(BOOL)intendedForPII {
+  NSDate *birthday = [_rewardedConnector userBirthday];
+  NSString *ageString = @"";
+
+  if (birthday) {
+    NSInteger ageInteger = [self ageFromBirthday:birthday];
+    ageString = [@"m_age:" stringByAppendingString:[@(ageInteger) stringValue]];
+  }
+
+  GADGender gender = [_rewardedConnector userGender];
+  NSString *genderString = @"";
+
+  if (gender == kGADGenderMale) {
+    genderString = @"m_gender:m";
+  } else if (gender == kGADGenderFemale) {
+    genderString = @"m_gender:f";
+  }
+  NSString *keywordsBuilder =
+      [NSString stringWithFormat:@"%@,%@,%@", kAdapterTpValue, ageString, genderString];
+
+  if (intendedForPII) {
+      if ([[MoPub sharedInstance] canCollectPersonalInfo]) {
+          return [self keywordsContainUserData:_rewardedConnector] ? keywordsBuilder : @"";
+      } else {
+          return @"";
+      }
+  } else {
+    return [self keywordsContainUserData:_rewardedConnector] ? @"" : keywordsBuilder;
+  }
+}
+
+- (NSInteger)ageFromBirthday:(NSDate *)birthdate {
+  NSDate *today = [NSDate date];
+  NSDateComponents *ageComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitYear
+                                                                    fromDate:birthdate
+                                                                      toDate:today
+                                                                     options:0];
+  return ageComponents.year;
+}
+
+- (BOOL)keywordsContainUserData:(id<GADMAdNetworkConnector>)rewardedConnector {
+  return [_rewardedConnector userGender] || [_rewardedConnector userBirthday] || [_rewardedConnector userHasLocation];
+}
+
+#pragma mark - Rewarded Video
+
+- (instancetype)initWithRewardBasedVideoAdNetworkConnector: (id)connector {
+    if (!connector) {
+        return nil;
+    }
+
+    _adExpired = false;
+    _rewardedConnector = connector;
+    _rewardedAdUnitId = [self.rewardedConnector.credentials objectForKey:@"pubid"];
+    
+    if ([_rewardedAdUnitId length] == 0) {
+        NSString *description = @"Failed to request a MoPub rewarded ad. Ad unit ID is empty.";
+        NSDictionary *userInfo = @{NSLocalizedDescriptionKey : description,
+                                   NSLocalizedFailureReasonErrorKey : description};
+        
+        NSError *error = [NSError errorWithDomain:kAdapterErrorDomain code:0 userInfo:userInfo];
+        [_rewardedConnector adapter:self didFailToLoadRewardBasedVideoAdwithError:error];
+    }
+
+    if (![[MoPub sharedInstance] isSdkInitialized]) {
+        [self initializeMoPub:_rewardedAdUnitId];
+    }
+
+    self = [super init];
+    if (self) {
+        [MPRewardedVideo setDelegate:self forAdUnitId:self.rewardedAdUnitId];
+    }
+    return self;
+}
+
+- (void)setUp {
+    [MPRewardedVideo setDelegate:self forAdUnitId:self.rewardedAdUnitId];
+    
```

Comment:
Also call `adapterDidSetUpRewardBasedVideoAd` here if the setup has succeeded. Like this 
` [_rewardedConnector adapterDidSetUpRewardBasedVideoAd:self];`